api/v1alpha1/selfnoderemediationconfig_webhook.go

```diff
@@ -0,0 +1,171 @@
+}
+
+// log is for logging in this package.
+var selfNodeRemediationConfigLog = logf.Log.WithName("selfnoderemediationconfig-resource")
+
+func (r *SelfNodeRemediationConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+
+	// check if OLM injected certs
+	certs := []string{filepath.Join(webhookCertDir, webhookCertName), filepath.Join(webhookCertDir, webhookKeyName)}
+	certsInjected := true
+	for _, fname := range certs {
+		if _, err := os.Stat(fname); err != nil {
+			certsInjected = false
+			break
+		}
+	}
+	if certsInjected {
+		server := mgr.GetWebhookServer()
+		server.CertDir = webhookCertDir
+		server.CertName = webhookCertName
+		server.KeyName = webhookKeyName
+	} else {
+		selfNodeRemediationConfigLog.Info("OLM injected certs for webhooks not found")
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-self-node-remediation-medik8s-io-v1alpha1-selfnoderemediationconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=self-node-remediation.medik8s.io,resources=selfnoderemediationconfigs,verbs=create;update,versions=v1alpha1,name=vselfnoderemediationconfig.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &SelfNodeRemediationConfig{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *SelfNodeRemediationConfig) ValidateCreate() error {
+	selfNodeRemediationConfigLog.Info("validate create", "name", r.Name)
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *SelfNodeRemediationConfig) ValidateUpdate(old runtime.Object) error {
+	selfNodeRemediationConfigLog.Info("validate update", "name", r.Name)
+
+	return r.validateTimes()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *SelfNodeRemediationConfig) ValidateDelete() error {
+	selfNodeRemediationConfigLog.Info("validate delete", "name", r.Name)
+
+	return nil
+}
+
+// validateTimes validates each time field in the PoisonPillConfig CR doesn't go below the minimum time
+// that was defined to it
+func (r *SelfNodeRemediationConfig) validateTimes() error {
+
+	s := r.Spec
+	fields := []field{
+		{s.PeerApiServerTimeout, MinDurPeerApiServerTimeout, ErrPeerApiServerTimeout},
+		{s.ApiServerTimeout, MinDurApiServerTimeout, ErrApiServerTimeout},
+		{s.PeerDialTimeout, MinDurPeerDialTimeout, ErrPeerDialTimeout},
+		{s.PeerRequestTimeout, MinDurPeerRequestTimeout, ErrPeerRequestTimeout},
+		{s.ApiCheckInterval, MinDurApiCheckInterval, ErrApiCheckInterval},
+		{s.PeerUpdateInterval, MinDurPeerUpdateInterval, ErrPeerUpdateInterval},
+	}
+
+	for _, f := range fields {
+		err := checkField(f.fieldDurValue.Milliseconds(), f.fieldMinValueString, f.errorMessage)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkField(inputTime int64, minValidDur string, errMessage string) error {
```

Comment:
You could make `checkField` a method of the `field` type, so you don't need to pass the arguments.
Maybe field.validate() would be a nice name then.